### PR TITLE
Molecular subtyping nbl

### PR DESCRIPTION
**Pull request by adilahiri**
#### What scientific question is your analysis addressing?
To molecularly subtype neuroblastoma, ganglioneuroblastoma, and ganglioneuroma samples into MYCN amplified or MYCN non-amplified.

#### What was your approach?
To obtain the NBL samples,  we first filtered the histology file based on `pathology_free_text_diagnosis` , `sample`, and 
`experimental_strategy`.  We only consider the following values in each column
| Pathology_Diagnosis                                                         |sample_type| experimental_strategy| 
|:----------------------------------------------------------------------------|----------:|---------------------:|
|Neuroblastoma                                                                | Tumor     |   WGS                |  
|Ganglioneuroblastoma                                                         |           |   WXS                | 
|Ganglioneuroblastoma, nodular                                                |           |Targeted Sequencing   |  
|Ganglioneuroblastoma, intermixed                                             |           |   RNA-Seq            | 
|Ganglioneuroma, maturing subtype OR Ganglioneuroblastoma, well differentiated|           |                      |

Next we filter the `consensus_wgs_plus_cnvkit_wxs.tsv.gz` and `gene-expression-rsem-tpm-collapsed.rds` for the gene symbol MYCN and then join it with the filtered histology file. We use this composite file to get the DNA and RNA biospecimen IDs for the records and then subtype them based on the following criteria:

**Subtyping criteria**:

**case 1**:
If `pathology_free_text_diagnosis` is **amplified** and `status` is **amplified** assign subtype as `NBL, MYCN amplified`

**case 2**:
If `pathology_free_text_diagnosis` is **non-amp** and `status` is **amplified** assign subtype as `NBL, MYCN amplified`

**case 3**:
If `pathology_free_text_diagnosis` is **non-amp** and `status` is **non-amp** assign subtype as `NBL, MYCN non-amplified`

**case 4**:
If `pathology_free_text_diagnosis` is **amplified** and status is **non-amp**
For such samples if there exists a TPM value, evaluate if the TPM is above or below the Suggested_Cutoff established in the image `results/TPM_Biospecimen_All_Samples_With_TMP.png` assign subtype as `NBL, MYCN amplified` or `NBL, MYCN non-amplified` respectively.  In case there is no TPM values then assign the subtype as `NA`. 

**case 5**:
If there are samples that are not yet subtyped but have a TPM value, assign them a subtype based on the Suggested_Cutoff.  

**case 6**:
Other remaining samples are not subtyped and the subtype field is left as `NA`.
#### What GitHub issue does your pull request address?
[Issue#417](https://github.com/PediatricOpenTargets/ticket-tracker/issues/417)

#### Which areas should receive a particularly close look?
The code and logic are explained throughout the `00-Analysis-RMD` file , additional information is also provided in the module README.  Please review the data filtering steps in Lines 77-224 and Lines 257-329. Basically, we would like to make sure we are not missing any NBL samples.  

In the plot `plot/TPM_Biospecimen_All_Samples_With_TMP.png`, we establish a `Suggested_Cutoff` for TPM values, we use this value for subtyping samples that fall under  case 4 and 5.  Please ensure if this cutoff is appropriate. 

Also review the results in the table [NBL_MYCN_Subtype.tsv](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/blob/molecular-subtyping-NBL/analyses/molecular-subtyping-NBL/results/NBL_MYCN_Subtype.tsv) and the QC
results in [QC_table.tsv](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/blob/molecular-subtyping-NBL/analyses/molecular-subtyping-NBL/results/QC_table.tsv)
#### Is there anything that you want to discuss further?
When finding samples that have both DNA and RNA IDs we encountered the following 2 issues with repeating records: 

1. Some of the biospecimen have same DNA and RNA IDs but differing copy numbers and status as mentioned in the following [Issue#436](https://github.com/PediatricOpenTargets/ticket-tracker/issues/436) and [comment](https://github.com/PediatricOpenTargets/ticket-tracker/issues/417#issuecomment-1281392325) , for the cases mentioned in the issue we retained the record with higher copy number. 

2. In addition we also found, two other repeating records with same DNA and RNA ID but differing aliquot_id
These records are 

 DNA_ID      | RNA_ID | aliquot_id  |
| :---        |    :----:   |          ---: |
|BS_0XC02E11| BS_2HM4AE24|ET_FD9T78QE_DGD_STNGS_29|
|BS_0XC02E11| BS_2HM4AE24|ET_FD9T78QE_DGD_STNGS_64|
| BS_1F8J25Q1 | BS_2HM4AE24 |ET_FD9T78QE_DGD_STNGS_29|
| BS_1F8J25Q1 | BS_2HM4AE24 |ET_FD9T78QE_DGD_STNGS_64|

We retained the copies with `aliquot_id` ET_FD9T78QE_DGD_STNGS_64, to tackle the issue of duplicates. Please review this and provide your feedback. The issues 1 and 2 are tackled in lines 196-224 in the `00-Analysis-RMD` file. 

`results/Alteration_Table.tsv`: This table is similar to the table `NBL_MYCN_Subtype.tsv`. However, this table has 
additional columns which contain information on `MYCN_TPM`,	`copy_number`,	`status`, and	`pathology_free_text_diagnosis`. Furthermore, the column `subtype` in this table provides more insights into the samples in `NBL_MYCN_Subtype.tsv` which had a subtype `NA` (samples which could not be subtyped). If samples fell into **case 4** but didn't have a TPM value, those are subtyped as `Pathology-amp,Status-non-amp,TPM-NA`.  If a sample did not fall into any of the above cases they are subtyped as `Unclassified due to insufficient info`.

#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?
Yes

#### What types of results are included (e.g., table, figure)?
Tables and Figures

#### What is your summary of the results?
The table  [NBL_MYCN_Subtype.tsv](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/blob/molecular-subtyping-NBL/analyses/molecular-subtyping-NBL/results/NBL_MYCN_Subtype.tsv)  has 1168 samples of which only 509 were assigned a subtype. 107 of the samples are non-amplified, 402 are amplified and the rest could not be subtyped due to missing information. 

#### Reproducibility Checklist

- [ ] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [X ] This analysis has been added to continuous integration.

#### Documentation Checklist

- [X ] This analysis module has a `README` and it is up to date.
- [X ] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [ X] The analytical code is documented and contains comments.

### CODE REVIEW

File: analyses/molecular-subtyping-NBL/00-Analysis.Rmd [hunk 1]
```diff
@@ -0,0 +1,633 @@
...
+```{r Data filtering}
+# First filter the histology file based on diseases we are intereseted in neuroblastoma, ganglioneuroblastoma, and ganglioneuroma
+hist_filtered_df <- hist_df %>% filter(pathology_diagnosis=="Neuroblastoma"|
+                                       pathology_diagnosis=="Ganglioneuroblastoma"|
+                                       pathology_diagnosis=="Ganglioneuroblastoma, nodular"|
+                                       pathology_diagnosis =="Ganglioneuroblastoma, intermixed"|
+                                       pathology_diagnosis ==
+                                         "Ganglioneuroma, maturing subtype OR Ganglioneuroblastoma, well differentiated",
+                                       sample_type=="Tumor",experimental_strategy %in%
+                                         c("WGS","WXS","Targeted Sequencing","RNA-Seq"))
```
Comment (sangeetashukla) [inline analyses/molecular-subtyping-NBL/00-Analysis.Rmd]:
```
hist_filtered_df <- hist_df %>% 
  filter(sample_type=="Tumor", 
         experimental_strategy %in% c("WGS", "WXS", "Targeted Sequencing", "RNA-Seq")) %>%
  filter(pathology_diagnosis== "Neuroblastoma" |
           pathology_diagnosis== "Ganglioneuroblastoma" |
           pathology_diagnosis== "Ganglioneuroblastoma, nodular" |
           pathology_diagnosis == "Ganglioneuroblastoma, intermixed" |
           pathology_diagnosis == "Ganglioneuroma, maturing subtype OR Ganglioneuroblastoma, well differentiated")
```

File: analyses/molecular-subtyping-NBL/00-Analysis.Rmd [hunk 2]
```diff
@@ -0,0 +1,633 @@
...
+and experimental strategy. The pathology_diagnosis can have certain values as listed in the code chunk below. 
+The sample type we query for is "tumor", i.e. we exclude "normal" samples. For experimental strategy, we consider
+WGS, WXS, Targeted Sequencing, and RNA-Seq. We ultimately want to create an output table with DNA and RNA 
+biospecimen IDs, with their corresponding molecular subtype, so we will need get DNA and RNA samples from the histology
+file. 
+
+We filter the consensus_wgs_plus_cnvkit_wxs.tsv.gz and gene-expression-rsem-tpm-collapsed.rds for MYCN to get the relevant information. 
+# Data Filtering
+```{r Data filtering}
+# First filter the histology file based on diseases we are intereseted in neuroblastoma, ganglioneuroblastoma, and ganglioneuroma
```
Comment (sangeetashukla) [inline analyses/molecular-subtyping-NBL/00-Analysis.Rmd]: # Extract tumor samples with treated with select experimental_strategy for NBL

**Comment by adilahiri** [comment]
@sangeetashukla: Thank you for the reviews here and during our meeting, I have condensed the code accordingly. I will look into breaking the code into two scripts as you suggested.

File: analyses/molecular-subtyping-NBL/01-Subtyping.Rmd [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+
+TPM_index <- c(index_clincal,index_tpm_unsubtype_above,index_tpm_unsubtype_below)
+
+df_tpm <- alteration_df[TPM_index,]
+
+df_tpm <- df_tpm %>% select(Kids_First_Biospecimen_ID_DNA,Kids_First_Biospecimen_ID_RNA,copy_number,
+                            pathology_free_text_diagnosis,status,MYCN_TPM,subtype)
+```
+
+# QC Check
```
Comment (jharenza) [inline analyses/molecular-subtyping-NBL/01-Subtyping.Rmd]: Please add some more comments here - what are you doing?

File: analyses/molecular-subtyping-NBL/01-Subtyping.Rmd [hunk 4]
```diff
@@ -0,0 +1,315 @@
+---
+title: "01-subtyping"
+author: "Aditya Lahiri, Eric Wafula, Jo Lynne Rokita"
+date: "11/14/2022"
+output: html_notebook
+---
+# Objective
+To molecularly subtype neuroblastoma, ganglioneuroblastoma, and ganglioneuroma samples into MYCN 
+amplified or MYCN non-amplified. This script loads the table `Alteration_Table.tsv` from the
```
Comment (jharenza) [inline analyses/molecular-subtyping-NBL/01-Subtyping.Rmd]: Question for @afarrel and JMM - do we want to separate the subtypes as "NBL, ganglioNB, and ganglioneuroma (some abbreviations)" or just leave all as "NBL, subtype"?

Comment (jharenza) [reply analyses/molecular-subtyping-NBL/01-Subtyping.Rmd]: @adilahiri some more context here - can you post a link to the original ticket, also mention that MYCN is on 2p (locaation) and that you will be assessing this with CNV calls, using clinically-reported MYCN status as a check, and additionally checking for consistency between TARGET and GMKF NBL samples?

File: analyses/molecular-subtyping-NBL/README.md [hunk 5]
```diff
@@ -0,0 +1,95 @@
+# Molecular Subtyping NBL
+**Module authors**: Aditya Lahiri ([@adilahiri](https://github.com/adilahiri)), Eric Wafula ([@ewafula](https://github.com/ewafula)), and Jo Lynne Rokita ([@jharenza](https://github.com/jharenza))
+
+**Module reviewers**: Sangeetha Shukla ([@sangeetashukla
+](https://github.com/sangeetashukla))
```
Comment (jharenza) [inline analyses/molecular-subtyping-NBL/README.md]:
```suggestion
```
not needed

File: analyses/molecular-subtyping-NBL/README.md [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+
+**case 3**:
+If `pathology_free_text_diagnosis` is **non-amp** and `status` is **non-amp** assign subtype as `NBL, MYCN non-amplified`
+
+**case 4**:
+If `pathology_free_text_diagnosis` is **amplified** and status is **non-amp**
+For such samples if there exists a TPM value, evaluate if the TPM is above or below the Suggested_Cutoff established in the image `results/TPM_Biospecimen_All_Samples_With_TMP.png` assign subtype as `NBL, MYCN amplified` or `NBL, MYCN non-amplified` respectively.  In case there is no TPM values then assign the subtype as `NA`. 
+
+**case 5**:
+If there are samples that are not yet subtyped but have a TPM value, assign them a subtype based on the Suggested_Cutoff.
```
Comment (jharenza) [inline analyses/molecular-subtyping-NBL/README.md]: if this cutoff is known now, can you document it here?

File: analyses/molecular-subtyping-NBL/README.md [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+
+**case 4**:
+If `pathology_free_text_diagnosis` is **amplified** and status is **non-amp**
+For such samples if there exists a TPM value, evaluate if the TPM is above or below the Suggested_Cutoff established in the image `results/TPM_Biospecimen_All_Samples_With_TMP.png` assign subtype as `NBL, MYCN amplified` or `NBL, MYCN non-amplified` respectively.  In case there is no TPM values then assign the subtype as `NA`. 
+
+**case 5**:
+If there are samples that are not yet subtyped but have a TPM value, assign them a subtype based on the Suggested_Cutoff.  
+
+**case 6**:
+Other remaining samples are not subtyped and the subtype field is left as `NA`.
```
Comment (jharenza) [inline analyses/molecular-subtyping-NBL/README.md]: This field should be left as `NBL, to be classified` if we cannot yet determine a definitive subtype. This lets us know we have some, but not enough information.

**Comment by ewafula** [comment]
> Hi @adilahiri - some general organizational comments:
> 
> * Can you remove the HTML READMEs? not sure why those are being added
> * If the file in `Intermediate` was derived from the module, just place in the results folder. No need for another folder. Also, if we can keep consistency of not using caps in filenames, that would be great. This file can be renamed `neuroblastoma_subtyping_relevant_data.tsv` to be consistent with other subtyping modules. You have this same table in `intermediate` and `results` (if they differ, we need to descriptively name the tables as such).
> * For the plots, can you rename these as `BSID_chr2p` or something more descriptive? Also, they are missing the x-axis label and the y-axis should be Log R Ratio, I believe, but please double check those units.
> * Can you rename `NBL_MYCN_Subtype.tsv` as `neuroblastoma_molecular_subtypes.tsv` or `nbl_molecular_subtypes.tsv` to be consistent with other subtyping modules and also rename the column `subtype` to `molecular_subtype` for easy digest into the histologies file downstream?
> * For tables with `status`, please make this more informative. This is `MYCN_CN_status`, for instance. Similarly, `copy_number` can be `MYCN_CN`.
> * In some of your files in which you derived
> * I agree with @sangeetashukla on breaking the scripts up but would ask that you go further on this - take a look at the embryonal module. There is a script to select the pathology diagnoses of interest, then another to subset files, then another to clean/prepare data, then one to prepare the final tables. Those would all typically be filed as separate PRs. I think we should update our repo readme to reflect some of the module creation and script organization expectations, @ewafula

Agree; we already discussed with @adilahiri following what was done in other subtyping modules and breaking up the code. My particular example had been EPN module, but the embryonal module is a good example too.

**Comment by adilahiri** [comment]
@jharenza thank you for the detailed feedback. I will incorporate the changes as you have suggested. 
I have removed the readme.html, in my latest updates, it was getting created as I was previewing the file before committing it.

**Review by jharenza — COMMENTED** [review]
@adilahiri can you clean up the results for this PR - I see a few files which seem to be the same but named slightly differently. (eg: "alteration_table_without_subtype" and "alteration_table_without_molecular_subtype"). I don't think we need the "without subtype" files. You also have two QC tables, one in caps one in lowercase. Let's aim to have minimal files for the PR - can you complete by end of week?

**Comment by ewafula** [comment]
@adilahiri,  `02-find-non-matching-biospecimen.Rmd` failing GA checks with the following error:
```
label: create the segmean vs location plots for samples where call is non-amp but pathology free text is amp
Quitting from lines 150-167 (03-subtyping.Rmd) 
Error: Faceting variables must have at least one value
```

**Comment by adilahiri** [comment]
@ewafula looking into the error, this  chunk of code was passing previously.

**Comment by adilahiri** [comment]
@jharenza: Thank you for your feedback. I have cleaned up the result folder and moved some of the intermediate files (tables) to the input directory. These intermediate files can deleted if required.  I have further updated the module readme and the individual scripts to include information regarding MYCN being on 2p, TPM cutoff and qc checks. The `NA` subtypes are relabeled to `NBL, to be classified`. 

@ewafula: Thank you for pointing out the GA errors and your feedback.

File: analyses/molecular-subtyping-NBL/00-subset-for-NBL.Rmd [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+
+hist_filtered_df <- hist_df %>% 
+  filter(sample_type=="Tumor", 
+         experimental_strategy %in% c("WGS", "WXS", "Targeted Sequencing", "RNA-Seq")) %>%
+  filter(pathology_diagnosis== "Neuroblastoma" |
+           pathology_diagnosis== "Ganglioneuroblastoma" |
+           pathology_diagnosis== "Ganglioneuroblastoma, nodular" |
+           pathology_diagnosis == "Ganglioneuroblastoma, intermixed" |
+           pathology_diagnosis == 
+           "Ganglioneuroma, maturing subtype OR Ganglioneuroblastoma, well differentiated")
```
Comment (jharenza) [inline analyses/molecular-subtyping-NBL/00-subset-for-NBL.Rmd]: Hi @adilahiri - just a few suggestions to clean this up a bit. Can you rework this to create a JSON file with a terms list, similar to the other modules such as [here](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/blob/dev/analyses/molecular-subtyping-HGG/00-HGG-select-pathology-dx.R)?

Comment (adilahiri) [reply analyses/molecular-subtyping-NBL/00-subset-for-NBL.Rmd]: Thank you for the feedback @jharenza, will get this done.

**Comment by ewafula** [comment]
@jharenza, @afarrel, this is ready to review.

**Review by afarrel — APPROVED** [review]
Thanks for working on this. Code looks much cleaner, results look reasonable after recent changes, the added code to fix the previously discussed issues looks okay. the code ran well on EC2.

**Pull request merged**